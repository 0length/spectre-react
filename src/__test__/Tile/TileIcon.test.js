import React from 'react'
import { shallow } from 'enzyme'

import { TileIcon } from '../..'

describe('TileIcon', () => {
  it('Should render default .tile-icon markup', () => {
    const wrapper = shallow(<TileIcon />)

    expect(wrapper.type()).toBe('div')
    expect(wrapper.hasClass('tile-icon')).toBe(true)
  })

  it('Should render children', () => {
    const wrapper = shallow(
      <TileIcon>
        <p>hello world</p>
      </TileIcon>
    )

    expect(wrapper.children().length).toBe(1)
    expect(wrapper.find('p').length).toBe(1)
  })

  it('Should pass classnames', () => {
    const wrapper = shallow(<TileIcon className='extra' />)

    expect(wrapper.hasClass('extra')).toBe(true)
    expect(wrapper.hasClass('tile-icon')).toBe(true)
  })

  it('Should render as a custom element', () => {
    const wrapper = shallow(<TileIcon renderAs='main'>custom</TileIcon>)

    expect(wrapper.text()).toBe('custom')
    expect(wrapper.hasClass('tile-icon')).toBe(true)
    expect(wrapper.type()).toBe('main')
  })

  it('Should match the snapshot', () => {
    const wrapper = shallow(<TileIcon />)

    expect(wrapper).toMatchSnapshot()
  })
})
