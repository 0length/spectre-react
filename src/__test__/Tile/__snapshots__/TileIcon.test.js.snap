// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileIcon Should match the snapshot 1`] = `
<div
  className="tile-icon"
/>
`;
